tasks/set_vars.yml

```diff
@@ -14,6 +14,8 @@
         - "{{ ansible_facts['distribution'] }}_{{ ansible_facts['distribution_version'] }}.yml"  # yamllint disable-line rule:line-length
         - "{{ ansible_facts['distribution'] }}_{{ ansible_facts['distribution_major_version'] }}.yml"  # yamllint disable-line rule:line-length
         - "{{ ansible_facts['distribution'] }}.yml"
+        - "{{ ansible_facts['os_family'] }}_{{ ansible_facts['distribution_version'] }}.yml"  # yamllint disable-line rule:line-length
+        - "{{ ansible_facts['os_family'] }}_{{ ansible_facts['distribution_major_version'] }}.yml"  # yamllint disable-line rule:line-length.yml"
```

Comment:
If this needs to be changed - I would rather use the method described at https://github.com/redhat-cop/automation-good-practices/tree/main/roles#supporting-multiple-distributions-and-versions than the "first_found" method currently used here.

Reply:
Running the role against a Rocky Linux 9 host results in the default settings file being used as there is no 'RedHat.yml'. It needs the RedHat_9.yml settings otherwise dracut complains about missing spaces around the kernel arguments in nbde_client.conf.
There's no obvious reason to have a Rocky_9.yml (or Alma... etc) as they all share the same config requirements.

Reply:
That would be fine as long as the RedHat.yml was generic enough, which comparing RedHat_7|8|9.yml that seems to be the case.

Reply:
> That would be fine as long as the RedHat.yml was generic enough,

Even if it isn't, you can override the settings from RedHat.yml with an e.g. RedHat_8.yml or a Rocky_8.yml, and then further refine those settings with e.g. a RedHat_8.4.yml or a Rocky_8.6.yml

> which comparing RedHat_7|8|9.yml that seems to be the case.

Reply:
The big problem I have with this is that `{{ ansible_facts['os_family'] }}_{{ ansible_facts['distribution_major_version'] }}.yml` is exactly the same as `{{ ansible_facts['distribution'] }}_{{ ansible_facts['distribution_major_version'] }}.yml` (and same for `distribution_version`)
If you want to provide a setting common to all os_family == RedHat-8, but have a different setting for distribution == RedHat-8, how to do that?

I think we will need different vars files for os_family - `vars/RedHat-os_family.yml` `vars/RedHat-8-os_family.yml` `vars/RedHat-8.8-os_family.yml`

The other problem is that some members of os_family RedHat are not strictly version number compatible with RHEL e.g. what should distribution CentOS do with the file `vars/RedHat-os_family-8.9.yml` since CentOS has no minor versions (at least, CentOS stream does not).  Same with Fedora - no minor version numbers, and the major version numbers are pretty far removed from RHEL version numbers.  I have no idea if Rocky, Alma, Oracle, etc. adhere strictly to the RHEL major.minor version numbering scheme - I suspect not since the end of publishing of the exact RHEL srpms.

So  `vars/RedHat-8-os_family.yml` will need to have the "lowest common denominator" settings that are applicable to all possible `8` versions of RHEL, CentOS, Rocky, Alma, Oracle, etc.